src/DelayId.cc

```diff
@@ -155,14 +138,13 @@ DelayId::bytesIn(int qty)
 
     assert ((unsigned short)(pool() - 1) != 0xFFFF);
 
-    if (compositeId != nullptr)
-        compositeId->bytesIn(qty);
+    compositeId->bytesIn(qty);
 }
 
 void
 DelayId::delayRead(const AsyncCall::Pointer &aRead)
 {
-    assert (compositeId != nullptr);
+    assert(*this);
```

Comment:
This new assertion is more strict than the current code. It is semantically correct, and I do not see any obvious violations in the current code, but I do see some inner delayRead() calls without immediate boolean operator checks and am not sure this PR should expand this assertion scope. I suggest leaving this code as it was.

Reply:
Ok.